src/modules/instruction-element/instruction-element.service.ts

```diff
@@ -91,4 +121,25 @@ export class InstructionElementService {
 
     await this.instructionElementRepository.delete(id)
   }
+
+  public async getGenericTypes(programId: number): Promise<InstructionElementGenericTypeMap> {
+    const accounts = await this.accountRepository.find({
+      where: {
+        program: {
+          id: programId,
+        },
+      },
+    })
+
+    const customTypes = accounts.map(account => ({
+      id: account.id,
+      name: account.name,
+      type: InstructionElementGenericType.CUSTOM_ACCOUNT,
+    }))
+
+    const instructionElementGenericTypeMap = new InstructionElementGenericTypeMap
+    instructionElementGenericTypeMap.custom_account_options.options = customTypes as []
```

Comment:
you can pass it as a parametr to the constructor new InstructionElementGenericTypeMap(customTypes)